ReSwift/CoreTypes/Store.swift

```diff
@@ -17,14 +17,26 @@ import Foundation
  */
 public class Store<State: StateType>: StoreType {
 
+    typealias SubscriptionType = Subscription<State>
+
     // TODO: Setter should not be public; need way for store enhancers to modify appState anyway
 
     /*private (set)*/ public var state: State! {
         didSet {
-            subscriptions.forEach {
+            var indexesToRemove: [Int] = []
+            for (index, subscription) in subscriptions.enumerate() {
+                guard let subscriber = subscription.subscriber else {
+                    indexesToRemove.append(index)
+                    return
```

Comment:
@Ben-G, I like how the code is concise with this approach, although I'm not sure if the best approach here is to run the filter on every attempt to notify the subscribers.